### PR TITLE
feat(formly-form): Use arrays for fieldTransforms

### LINKED ISSUE #458: Make fieldTransform an array.
This would make it possible to have multiple `fieldTransforms`.

This could be implemented as a non-breaking change by making this capable of handling a function or an array of functions: https://github.com/formly-js/angular-formly/blob/55cbe3c1288daae1ea0a0f18c99834271ecced96/src/directives/formly-form.js#L138-L143

We'd want to log a deprecation warning, add error/warning docs, and test that (like we've done before, [implementation](https://github.com/formly-js/angular-formly/blob/v6.26.1/src/providers/formlyConfig.js#L271-L278) | [warning docs](https://github.com/formly-js/angular-formly/blob/v6.26.1/other/ERRORS_AND_WARNINGS.md#apicheck-as-an-object-deprecated) | [test](https://github.com/formly-js/angular-formly/blob/v6.26.1/src/providers/formlyConfig.test.js#L610-L621)).

Then in a breaking release we could remove that logic and require it to be an array.

This one's up for grabs (whether you've contributed to open source before or not!)

### PR DESCRIPTION
**Pull request by BarryThePenguin**
Documentation and deprecation warnings for both options and formConfig.extras
Fixes #458

### CODE REVIEW

**Comment by BarryThePenguin** [comment]
I might add some more tests when I get time

Another thought I had was specifying fieldTransforms as a keyed object, that way you can have specific transforms applied to different fields...

```js
formConfig.extras.fieldTransforms = {
    transformOne: () => { },
    transformTwo: () => { },
};

...

fields = [{
    name: 'one',
    fieldTransforms: ['transformOne']
}, {
    name: 'two',
    fieldTransforms: ['transformTwo']
}]
```

**Comment by kentcdodds** [comment]
Thanks for this. Can't merge now, but will look more closely later. The problem with object is order isn't consistent/garunteed... Though I like the idea, I think we'll wait for someone to have aause case before we implement something like that.

Thanks again! I'll look at this later. In the mean time, could I ask for your feedback on ease of contributing?

File: src/directives/formly-form.js [hunk 1]
```diff
@@ -133,21 +133,58 @@ function formlyForm(formlyUsability, formlyWarn, $parse, formlyConfig, $interpol
...
+        }
+      });
+
       setupModels();
 
       angular.forEach($scope.fields, attachKey); // attaches a key based on the index if a key isn't specified
       angular.forEach($scope.fields, setupWatchers); // setup watchers for all fields
     }
 
+    function warn() {
```
Comment (kentcdodds) [inline src/directives/formly-form.js]: Is there a reason you didn't just inject the [`formlyWarn`](https://github.com/formly-js/angular-formly/blob/master/src/services/formlyWarn.js) service and use that instead?

Comment (BarryThePenguin) [inline src/directives/formly-form.js]: No reason, just didn't see it there. Will switch it out

Comment (kentcdodds) [inline src/directives/formly-form.js]: That's impressive that you created almost exactly what I had. Haha, I thought it was copy/paste code 😀

**Comment by kentcdodds** [comment]
Odd... Travis says that the build passed, but if I look at the output, it shows that two tests failed...

**Comment by BarryThePenguin** [comment]
Hmm... I do have ghooks disabled (on Windows). Will look at it later. I'm in no hurry to get this merged, just happy to contribute

**Comment by kentcdodds** [comment]
I'm happy to have you contribute! Looks like they're failing locally. I'll try to see what's going on. I'll let you know. Thanks for contributing :-)

**Comment by kentcdodds** [comment]
I've found the problem. I'll submit a PR to your branch :-)

**Comment by kentcdodds** [comment]
Perfect! Thank you!

**Comment by BarryThePenguin** [comment]
Thanks for the assistance! I'll try some feedback...

I find structure really helpful, code style, contribution guidelines, have always been helpful when I've contributed to projects in the past. Since using formly and seeing first-timers-only gain momentum, it's only recently I've realised how helpful these ideas are for the community.

Using ghooks to run linting, tests, etc is really helpful. Saves all the nitpicking 'why did you add that tab/space/symbol/emoji'. Super beneficial. The downside is making it work in everyone's dev environment/setup. How can we improve that experience for people, especially when the node/npm world is so modular and projects have so many dependencies, for example commitizen/cz-cli#10.

How much hand-holding should there be? I think it's enough to reply to people's queries with a quick link or short description of where to find the solution. From what I've seen in Gitter, you do a good job.

From this side of the world it looks like you're making good things. Look after yourself

**Comment by kentcdodds** [comment]
:-D thanks for the feedback! I'm glad that overall it was a good
experience. Thanks again for contributing. Let's do this again sometime :-D

- Kent C. Dodds

(Sent from my mobile device, please forgive typos or brevity)

**Pull request merged**